src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs

```diff
@@ -371,6 +397,20 @@ private IEnumerable<PackageReference> GetPackageReferences(IEnumerable<LibraryDe
                 .Select(library => new BuildIntegratedPackageReference(library, targetFramework, GetPackageReferenceUtility.UpdateResolvedVersion(library, targetFramework, assetsPackageSpec?.TargetFrameworks.FirstOrDefault(), targets, installedPackages)));
         }
 
+        private async Task<IEnumerable<PackageReference>> GetTransitivePackageReferencesAsync(PackageSpec packageSpec)
+        {
+            await GetPackageReferencesAsync(packageSpec);
+            return _transitivePackages.Values;
+        }
+
+        private IEnumerable<PackageReference> GetTransitivePackageReferences(IEnumerable<PackageDependency> libraries, NuGetFramework targetFramework, PackageSpec assetsPackageSpec, IList<LockFileTarget> targets)
+        {
+            IEnumerable<PackageReference> transitive = new List<PackageReference>();
+            transitive = libraries
+                .Select(library => new PackageReference(GetPackageReferenceUtility.GetResolvedVersionTransitive(library, targetFramework, targets), targetFramework));
```

Comment:
Wouldn't this include the direct dependencies as well?

Reply:
I get these packages from the 'dependencies' sections in the assets file. So they won't include all of the top level packages. An exception would be if a package is a top level packages as well as a dependency for another package. In that case it would be included in both sets. 
Looking at this again, I see that I could instead just grab all of the packages in the 'libraries' section of the assets file. It looks like then I would get all packages (top level plus transitive). 
Do you have a preference? Either will work for the recommender.

Reply:
I think the transitive section should be as correct as possible. So if a package is a direct dependency, thne it shouldn't be contained in the transitive packages list. 
Going through libraries might be easier. 

Remember to exclude projects.

Reply:
Added code to filter out packages that are already included in the top level packages.